src/CommandLineUtils/MappedOption{T}.cs

```diff
@@ -0,0 +1,187 @@
+        /// </summary>
+        protected readonly CommandLineApplication _commandLineApplication;
+        private readonly List<ConstantValueOption<T>> _mappings;
+        private readonly ICollection<IOptionValidator> _validators;
+
+        /// <summary>
+        /// Create a new set of mapped options
+        /// </summary>
+        /// <param name="commandLineApplication">the application this option is for</param>
+        /// <param name="optionType">the option type</param>
+        /// <exception cref="ArgumentException">if optionType is something other than <see cref="CommandOptionType.MultipleValue"/> or <see cref="CommandOptionType.SingleValue"/></exception>
+        public MappedOption(CommandLineApplication commandLineApplication, CommandOptionType optionType)
+        {
+            if (optionType != CommandOptionType.MultipleValue && optionType != CommandOptionType.SingleValue)
+            {
+                throw new ArgumentException("Only MultipleValue or SingleValue is supported.", nameof(optionType));
+            }
+
+            _commandLineApplication = commandLineApplication;
+            _mappings = new List<ConstantValueOption<T>>();
+            _validators = new List<IOptionValidator>();
+            OptionType = optionType;
+
+            if (optionType == CommandOptionType.SingleValue)
+            {
+                _validators.Add(new SingleValueValidator());
+            }
+        }
+
+        /// <summary>
+        /// Get's all the mapped values as <see cref="ConstantValueOption{T}"/>
+        /// </summary>
+        public IReadOnlyList<ConstantValueOption<T>> Mappings => _mappings;
+
+        /// <summary>
+        /// Add mapping
+        /// </summary>
+        /// <param name="template">parameter-template e.g. <code>-a|--apple</code></param>
+        /// <param name="value">value to use if the mapped parameter was specified</param>
+        /// <returns><code>this</code></returns>
+        public MappedOption<T> Map(string template, T value)
+        {
+            Add(template, value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add mapping
+        /// </summary>
+        /// <param name="description">description for use in the help-text</param>
+        /// <param name="template">parameter-template e.g. <code>-a|--apple</code></param>
+        /// <param name="value">value to use if the mapped parameter was specified</param>
+        /// <returns><code>this</code></returns>
+        public MappedOption<T> Map(string description, string template, T value)
+        {
+            Add(description, template, value);
+            return this;
+        }
+
+        internal ConstantValueOption<T> Add(T value)
+        {
+            var constantValueOption = new ConstantValueOption<T>(value);
+            _mappings.Add(constantValueOption);
+            _commandLineApplication.AddOption(constantValueOption);
```

Comment:
The tricky thing for me was adding it to both places. I didn't want to change the parsing logic to take a list of strings for each argument style (short, long, symbol) - which I thought would be required to otherwise ensure that each possible flow of actions would lead to a working construction.